### PR TITLE
Adaptor Openeuler 22.03-lts-sp2

### PR DESCRIPTION
**Pull request by peter5232**
### What problem does this PR solve?

Issue Number: #2612 

Problem Summary:

### What is changed and how it works?

What's Changed:

How it Works:

Side effects(Breaking backward compatibility? Performance regression?):

### Check List

- [ ] Relevant documentation/comments is changed or added
- [x] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

**Comment by peter5232** [comment]
cicheck

**Comment by peter5232** [comment]
cicheck

File: curvefs/docker/openeuler/entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env bash
+
+# Copyright (C) 2021 Jingli Chen (Wine93), NetEase Inc.
```
Comment (Cyber-SiKu) [inline curvefs/docker/openeuler/entrypoint.sh:3]: time and signature

File: docker/openeuler/entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env bash
+
+# Copyright (C) 2021 Jingli Chen (Wine93), NetEase Inc.
```
Comment (Cyber-SiKu) [inline docker/openeuler/entrypoint.sh:3]: ditto

**Review by Cyber-SiKu — COMMENTED** [review]
You can change the default os to openeuler, and I will bring it back to help you package and upload the base image, and you can run ci again (just leave a successful record) and then change it back.

File: docker/openeuler/curve-tgt/Makefile [hunk 3]
```diff
@@ -0,0 +1,7 @@
+.PHONY: build
+
+curve-sdk:
+	docker build --target  curve-tgt -t opencurvedocker/curve-base:curve-tgt-openeuler .
```
Comment (Cyber-SiKu) [inline docker/openeuler/curve-tgt/Makefile:4]: https://github.com/opencurve/curve/pull/2733

**Comment by peter5232** [comment]
cicheck

**Review by Cyber-SiKu — APPROVED** [review]

**Comment by peter5232** [comment]
cicheck

**Comment by peter5232** [comment]
cicheck

**Comment by peter5232** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
cicheck

**Comment by peter5232** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
cicheck

**Comment by peter5232** [comment]
cicheck

**Comment by peter5232** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
cicheck

File: docker/openeuler/compile/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    PROTOC_GEN_GO_VERSION=v1.28 \
+    PROTOC_GEN_GO_GRPC_VERSION=v1.2
+RUN yum update -y\
+    && yum install -y gcc gcc-c++ bison bison-devel flex flex-devel cmake make wget curl git protobuf-compiler patch libevent-devel musl-devel\
+    && yum install -y unzip which zlib zlib-devel openssl openssl-devel libnl3 libnl3-devel libuuid libuuid-devel libcurl-devel boost boost-devel wget cmake tar  \
+    && yum install -y libunwind libunwind-devel \
+    && wget https://github.com/bazelbuild/bazel/releases/download/4.2.2/bazel-4.2.2-installer-linux-x86_64.sh && bash bazel-4.2.2-installer-linux-x86_64.sh \
+    && wget https://curve-build.nos-eastchina1.126.net/libfiu-1.00.tar.gz && tar -zxvf libfiu-1.00.tar.gz && cd libfiu-1.00 && make libfiu && make libfiu_install && cd .. && rm -rf libfiu-1.00 \
+    &&wget https://golang.google.cn/dl/go1.20.6.linux-amd64.tar.gz && tar xzvf go1.20.6.linux-amd64.tar.gz && echo "export PATH=\$PATH:/go/bin" >> ~/.bashrc \
+   && wget https://curve-build.nos-eastchina1.126.net/memcache/memcached-1.6.17.tar.gz \
```
Comment (wu-hanqing) [inline docker/openeuler/compile/Dockerfile:17]: indent alignment

File: docker/openeuler/compile/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    && tar xvf memcached-1.6.17.tar.gz && cd memcached-1.6.17/ \
+    && ./configure && make -j&& make install \
+    && cd .. && rm -rf memcached-1.6.17 memcached-1.6.17.tar.gz \
+    && yum install -y fuse3 fuse3-devel lz4-devel snappy-devel\
+   && echo "export LD_LIBRARY_PATH=\$LD_LIBRARY_PATH:/usr/local/lib" >> ~/.bashrc \
+   && source ~/.bashrc
```
Comment (wu-hanqing) [inline docker/openeuler/compile/Dockerfile:23]: ditto

**Review by wu-hanqing — COMMENTED** [review]
It looks like there are a lot of scripts and dockerfiles that are duplicates with debian11. Is there some way to merge these duplicate files?

**Comment by peter5232** [comment]
cicheck

**Comment by peter5232** [comment]
cicheck

**Comment by peter5232** [comment]
cicheck

**Comment by peter5232** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
cicheck

**Comment by peter5232** [comment]
cicheck

**Comment by peter5232** [comment]
cicheck

**Comment by peter5232** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
cicheck

**Comment by peter5232** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
cicheck

**Comment by peter5232** [comment]
cicheck

**Comment by peter5232** [comment]
cicheck

**Comment by peter5232** [comment]
cicheck

**Comment by peter5232** [comment]
cicheck

**Comment by peter5232** [comment]
cicheck

**Comment by peter5232** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
replace by [pr](https://github.com/opencurve/curve/pull/2745)

**Pull request closed without merging**